chargeCurrentPairsFinder.c

```diff
@@ -0,0 +1,88 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "chargeCurrentPairsFinder.h"
+
+int ascendingComparator(const void * a, const void * b) 
+{
+   return ( *(int*)a - *(int*)b );
+}
+
+static int validateCurrentSampleArray(int *chrgeCurrentArray, int size)
+{
+    if(size < 2)
+    {
+        return 0;
+    }
+
+    if(chrgeCurrentArray == NULL)
+    {
+        return 0;
+    }
+
+    return 1;
+}
+
+static int validateCurrentPair(int index, int firstElement, int lastElement)
+{
+    if(firstElement == lastElement)
+    {
+        return 0;
+    }
+
+    return index;
+}
+
+static int identifyCurrentPair(int *chrgeCurrentArray, int size)
```

Comment:
Using static functions should be done carefully, because in case of multithreading sharing of data could lead to unexpected behaviur and also difficulty in debugging.

Reply:
I intentionally used static keyword to restrict the scope of the functions to the particular module alone. It doesn't share any data. I don't use any global or static variables between functions which could cause issues when used in multi threading. 

Please correct me if I am wrong :)